src/components/map-viewer.jsx

```diff
@@ -128,212 +131,254 @@ const MapViewer = ({
         onChangeTab(1); // switch to spreadsheet view
     }
 
+    const [instructionSnakbar, setInstructionSnackbar] = useState(undefined);
     useEffect(() => {
         //display a snackbar
-        if (isDrawingMode) {
-            snackInfo({
-                messageTxt: intl.formatMessage({
-                    id: 'DrawingPolygonInstruction',
-                }),
-            });
+        if (isDrawingMode && !instructionSnakbar) {
+            setInstructionSnackbar(
+                snackInfo({
+                    messageTxt: intl.formatMessage({
+                        id: 'DrawingPolygonInstruction',
+                    }),
+                    persist: true,
+                })
+            );
         }
-    }, [isDrawingMode, intl, snackInfo]);
-
-    function getMapWitdh(displayMode) {
-        if (displayMode === STUDY_DISPLAY_MODE.DRAW) {
-            return '80%';
-        } else {
-            return '100%';
+        if (!isDrawingMode && instructionSnakbar) {
+            closeSnackbar(instructionSnakbar);
+            setInstructionSnackbar(undefined);
         }
-    }
+    }, [isDrawingMode, intl, snackInfo, instructionSnakbar, closeSnackbar]);
+
+    const onDrawingModeEnter = useCallback(
+        (active) => {
+            // save the previous mode so we can restore it when the user cancel the drawing
+            if (previousStudyDisplayMode.current === undefined) {
+                previousStudyDisplayMode.current = studyDisplayMode;
+            }
+            if (active === true) {
+                dispatch(setStudyDisplayMode(STUDY_DISPLAY_MODE.MAP));
+            }
+            setIsDrawingMode(active);
+        },
+        [dispatch, previousStudyDisplayMode, studyDisplayMode]
+    );
+
     const onCancelFunction = useCallback(() => {
         networkMapref.current.cleanDraw();
-        dispatch(setStudyDisplayMode(STUDY_DISPLAY_MODE.MAP));
-    }, [dispatch]);
+        if (previousStudyDisplayMode.current !== undefined) {
+            dispatch(setStudyDisplayMode(previousStudyDisplayMode.current));
+            previousStudyDisplayMode.current = undefined;
+        }
+    }, [dispatch, previousStudyDisplayMode]);
+
     return (
-        <ReactFlowProvider>
-            <Box sx={styles.table}>
-                <Box sx={styles.horizontalToolbar}>
-                    <HorizontalToolbar />
-                </Box>
+        <Box sx={styles.table}>
+            <Box sx={styles.horizontalToolbar}>
+                <HorizontalToolbar />
+            </Box>
+            <div
```

Comment:
Why mix div and box? Can't we use Box to use the sw attributes and put the style at the top of the file?